src/main/java/org/cascadebot/cascadebot/music/KaraokeHandler.java

```diff
@@ -0,0 +1,112 @@
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class KaraokeHandler {
+
+    private static Map<Long, Boolean> karaokeEnabled = new HashMap<>();
+
+    public static boolean isKaraoke(Long guildId) {
+        if (karaokeEnabled.get(guildId) == null) {
+            return false;
+        } else { return karaokeEnabled.get(guildId); }
+    }
+
+    public static void setKaraoke(Long guildId, Boolean status) {
+        karaokeEnabled.put(guildId, status);
+    }
+
+    public static void getSongLyrics(String trackId, TextChannel channel, Long guildId, Message message) throws ParserConfigurationException {
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        factory.setValidating(true);
+        factory.setIgnoringElementContentWhitespace(true);
+        DocumentBuilder builder = factory.newDocumentBuilder();
+        Request request = new Request.Builder().url("https://video.google.com/timedtext?lang=en&v=" + trackId).build();
+        CascadeBot.INS.getHttpClient().newCall(request).enqueue(new Callback() {
+            @Override
+            public void onFailure(@NotNull Call call, @NotNull IOException e) {
+                Messaging.sendExceptionMessage(channel, Language.i18n(channel.getGuild().getIdLong(), ""), e);
+            }
+
+            @Override
+            public void onResponse(@NotNull Call call, @NotNull Response response) {
+
+                if (response.body() == null) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                if (response.code() != 200) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                String body = "";
+                try {
+                    body = response.body().string();
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+
+                if (body.equals("")) {
+                    Messaging.sendDangerMessage(channel, Language.i18n(channel.getGuild().getIdLong(), "commands.karaoke.cannot_find"));
+                    return;
+                }
+
+                try {
+                    Document doc = builder.parse(new StringInputStream(body));
+                    NodeList texts = doc.getElementsByTagName("text");
+                    Captions captions = new Captions();
+                    for (int i = 0; i < texts.getLength(); i++) {
+                        NamedNodeMap attrs = texts.item(i).getAttributes();
```

Comment:
```suggestion

                        NamedNodeMap attrs = texts.item(i).getAttributes();
```

Reply:
The request is to add a newline? Why?

Reply:
Too clogged up, in my opinion

Reply:
I don't think this is necessary, second opinion please?

Reply:
Not necessary and would go against our global code style.